crates/floresta-chain/src/pruned_utreexo/partial_chain.rs

```diff
@@ -426,13 +382,30 @@ impl BlockchainInterface for PartialChainState {
 
     fn validate_block(
         &self,
-        _block: &bitcoin::Block,
+        block: &bitcoin::Block,
         _proof: rustreexo::accumulator::proof::Proof,
-        _inputs: HashMap<bitcoin::OutPoint, bitcoin::TxOut>,
+        inputs: HashMap<bitcoin::OutPoint, bitcoin::TxOut>,
         _del_hashes: Vec<bitcoin::hashes::sha256::Hash>,
-        _acc: Stump,
+        validate_block: bool,
     ) -> Result<(), Self::Error> {
-        unimplemented!("PartialChainState::validate_block")
+        let height = self.inner().current_height;
+        let flags = self.inner().get_validation_flags(height);
+        let ancestor = match self.inner().get_block(height) {
+            Some(h) => h,
+            _ => {
+                return Err(BlockchainError::BlockValidation(
+                    BlockValidationErrors::BlockExtendsAnOrphanChain,
+                ))
+            }
+        };
+        self.inner().consensus.validate_block(
```

Comment:
nit: bind inner to a variable and don't call `inner()`.  I find that way easier to read

```rust
let inner = self.inner();
inner.get_validation_flags(height);
....
```